src/main/java/org/skypro/examgen/service/ExaminerServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package org.skypro.examgen.service;
+
+import org.skypro.examgen.exception.NotEnoughQuestionsException;
+import org.skypro.examgen.model.Question;
+import org.springframework.stereotype.Service;
+import java.util.*;
+
+@Service
+public class ExaminerServiceImpl implements ExaminerService {
+    private final JavaQuestionService javaQuestionService;
```

Comment:
не забываем про полиморфизм
```suggestion
    private final QuestionService javaQuestionService;
```